src/metadata.ts

```diff
@@ -15,13 +16,14 @@ export const generateServiceProviderMetadata = (
     issuer,
     callbackUrl,
     logoutCallbackUrl,
-    identifierFormat,
-    wantAssertionsSigned,
     decryptionPvk,
     privateKey,
     metadataContactPerson,
     metadataOrganization,
-    generateUniqueId,
+    // The below defaults are selected to match the defaults in the `SAML` class.
+    identifierFormat = "urn:oasis:names:tc:SAML:1.1:nameid-format:emailAddress",
```

Comment:
I'd prefer to keep all the defaults in a single location. If they need to be duplicated here, then perhaps we should put a run-time check to make sure they are populated and leave the onus on the one calling this method directly to know what their doing.

Reply:
Is there any particular way you'd like me to share these defaults? Maybe declare them in a constant somewhere? Is there precedent for this?

I'd rather not leave this to the caller; if I'm not explicitly configuring `identifierFormat` for instances of `SAML`, I wouldn't expect to have to configure them here either.

Reply:
I think I might be OK with a set of constants in a new file: `constants.ts`. I see we've already polluted the `types.ts` file a little with a class that should probably be in `utilities.ts`, but I don't want that to be a pattern. It also doesn't seem to be a great fit for `utilities.ts`. I'm interested in your thoughts.

I really want to make sure that the next guy (or future me) doesn't think there is just one place for defaults and forgets to change something.

Reply:
I think your motivation is spot on. I think a `constants.ts` file with exports like `DEFAULT_IDENTIFIER_FORMAT` and `DEFAULT_WANT_ASSERTIONS_SIGNED` would be a good idea; I'll make that change if you're good with it!

Reply:
@cjbarth done in commit d38a77d.